lib/collection/src/update_handler.rs

```diff
@@ -465,6 +465,12 @@ impl UpdateHandler {
         // Check if Qdrant triggered any optimizations since starting at all
         let has_triggered_any_optimizers = self.has_triggered_optimizers.load(Ordering::Relaxed);
 
+        if self.max_optimization_threads == Some(0) {
+            // Collection cannot have optimization threads. So we cannot have suboptimal optimizers
+            // This eventually leads to ShardStatus::Green
+            return (has_triggered_any_optimizers, false);
+        }
```

Comment:
Just return `true` here for `has_triggered_any_optimizers` (I think).

In this specific case it doesn't mater whether optimizers are triggered once or not, and the current implementation still shows an intermediate Grey status.

I think it's better to show Green right away.

```suggestion
        if self.max_optimization_threads == Some(0) {
            // Collection cannot have optimization threads. So we cannot have suboptimal optimizers
            // This eventually leads to ShardStatus::Green
            return (true, false);
        }
```

Reply:
> and the current implementation still shows an intermediate Grey status

I don't think so. How? :thinking: 

> Just return true here for has_triggered_any_optimizers

I think they are independent variables. `has_triggered_any_optimizers` can be both true or false even it doesn't matter for now because `has_suboptimal_optimizers = false`.